src/core/mutatus.scala

```diff
@@ -30,25 +33,61 @@ object `package` {
     DatastoreOptions.getDefaultInstance.getService
   )
 
+  /** provides `saveAll` and `deleteAll` methods for collections of case class instances */
+  implicit class DataBatchExt[T <: Product](values: Iterable[T]) {
+
+    /** saves the all case class as a Datastore entity in batch mode */
+    def saveAll()(implicit svc: Service,
+                  encoder: Encoder[T],
+                  dao: Dao[T],
+                  idField: IdField[T]): Map[T, Ref[T]] = {
+      val (batch, refs) =
+        values.foldLeft(svc.readWrite.newBatch() -> mutable.Map.empty[T, Ref[T]]) {
```

Comment:
Is there a reason why you're using a mutable `Map` here, rather than an immutable one?

Reply:
Only due to overhead of creating new map for each entity in fold. I though that in case of large collections it may be better to use mutable variant. Although in most cases collections would be small, so we can use immutable Map without problems.

Reply:
It's a bit of an unknown. I think I'd prefer to see an immutable map there. Any performance issues we have will most likely be I/O-related.